internal/slicer/report.go

```diff
@@ -0,0 +1,47 @@
+package slicer
+
+import (
+	"io/fs"
+
+	"github.com/canonical/chisel/internal/fsutil"
+	"github.com/canonical/chisel/internal/setup"
+)
+
+type ReportEntry struct {
+	Path   string
+	Mode   fs.FileMode
+	Hash   string
+	Size   int
+	Slices []*setup.Slice
+	Link   string
+}
+
+// Report holds the information about files and directories created when slicing
+// packages.
+type Report struct {
+	Root string
```

Comment:
nit: Let's please add a comment about `Root`. I am assuming it's the directory where the report will be generated.

Reply:
Root is the root of the created `fs`, it will be used later by the report to trim the paths to get the relative to said root. The directory where `chisel.db` will be placed will be an argument to the output function not yet created.

Reply:
Rafid's suggestion is to have a comment in the code:
```
// Root is the filesystem path where the all reported content is based.
```